Tests/Functional/app/AppKernel.php

```diff
@@ -1,29 +1,43 @@
 <?php
 
-use Payum\Bundle\PayumBundle\PayumBundle;
-use Payum\Bundle\PayumBundle\Tests\Functional\app\AppKernelShared;
-use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
-use Symfony\Bundle\TwigBundle\TwigBundle;
 use Symfony\Component\Config\Loader\LoaderInterface;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\Kernel;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpKernel\HttpKernelInterface;
 
-if (Kernel::MAJOR_VERSION === 4) {
-    class AppKernel extends AppKernelShared
+class AppKernel extends Kernel
+{
+    public function registerBundles(): iterable
     {
-        public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQUEST, $catch = true): Response
-        {
-            return parent::handle($request, $type, false);
-        }
+        $bundles = array(
+            new Symfony\Bundle\FrameworkBundle\FrameworkBundle(),
+            new Symfony\Bundle\TwigBundle\TwigBundle(),
+            new Payum\Bundle\PayumBundle\PayumBundle(),
+        );
+
+        return $bundles;
+    }
+
+    public function getCacheDir(): string
+    {
+        return sys_get_temp_dir() . '/PayumBundle/cache';
+    }
+
+    public function getLogDir(): string
+    {
+        return sys_get_temp_dir() . '/PayumBundle/logs';
+    }
+
+    public function registerContainerConfiguration(LoaderInterface $loader): void
+    {
+        $loader->load(__DIR__ . '/config/config.yml');
+
+        $loader->load(__DIR__ . '/config/config_sf' . Kernel::MAJOR_VERSION . '.yml');
     }
-} else {
-    class AppKernel extends AppKernelShared
+
+    public function handle(Request $request, int $type = HttpKernelInterface::MAIN_REQUEST, bool $catch = true): Response
```

Comment:
Is this method needed if it just calls the parent method?

Reply:
The difference is the `$catch` parameter.

If i remove the function 7 tests will fail.
`[error] Uncaught PHP Exception Symfony\Component\HttpKernel\Exception\NotFoundHttpException: "A token with hash `payum_token` could not be found." at /Users/georg/PhpStorm/open-source/_backup/PayumBundle/vendor/payum/core/Payum/Core/Bridge/Symfony/Security/HttpRequestVerifier.php line 49`

The tests have a reference to @ticket 507 I guess https://github.com/Payum/PayumBundle/issues/507

I am not sure if this tests are useful anymore, and could mabye removed.
This needs a deeper look.